docs/userguide/campaigns.md

```diff
@@ -1,25 +1,125 @@
 
-## How to use campaigns
+When constructing a campaign, it is necessary to provide two objects:
+A search space (see [class](baybe.searchspace.core.SearchSpace) resp. [user guide](./searchspace))
+and an optimization objective (see [class](baybe.objective.Objective) resp. [user guide](./objective)).
 
-We have a detailed example explaining how to use the ``Campaign`` object in our examples, so please see [here](./../../examples/Basics/campaign) for all necessary information.
\ No newline at end of file
+The search space describes the possible configuration of parameters that can be tested during the campaign.
+
+```python
+from baybe.parameters import CategoricalParameter, NumericalDiscreteParameter
+from baybe.searchspace import SearchSpace
+
+grind_size = CategoricalParameter(name="grind_size", values=["coarse", "rather_coarse", "rather_fine", "fine"])
+temp = NumericalDiscreteParameter(name="temp", values = tuple([t/10.0 for t in range (905, 961)]))
+time = NumericalDiscreteParameter(name="time", values = tuple(range(120, 370, 10)))
+
+space = SearchSpace.from_product([grind_size, temp, time])
+```
+
+We refer to the [parameters](./parameters) resp. [search spaces](./searchspace) user
+guides for more details.
+
+To assess the quality of our coffee, we use a subjective evaluation method
+involving tasting and assigning a rating on a scale of 1 to 10, where a higher score
+indicates better quality.
+
+```python
+from baybe.targets import NumericalTarget
+from baybe.objective import Objective
+
+target = NumericalTarget(name="quality", mode="MAX", bounds=(1,10))
+objective = Objective(mode="SINGLE", targets=[target])
+```
+
+Combining the search space and the objective now enables us to create a campaign.
+
+```python
+from baybe.campaign import Campaign
+
+campaign = Campaign(searchspace=space, objective=objective)
+```
+
+## Getting a recommendation and adding a measurement
+
+To obtain a recommendation for the next experiment, we can query the campaign and use
+the [`recommend`](baybe.campaign.Campaign.recommend) function. The function takes only
+one argument, which is the `batch_quantity` keyword. This specifies the desired size of
+the batch of experiments to be conducted.
+
+```python
+rec = campaign.recommend(batch_quantity=3)
+```
+
+The `recommend` function returns a `DataFrame` with `batch_quantity` many rows, each 
+representing a set of parameters from the search space.
+To add measurements, we expand the `DataFrame` by adding a new column for the target.
+We can then provide the campaign with these measurements and receive a new
+recommendation.
+
+```python
+rec["quality"] = [2,4,9]
+campaign.add_measurements(rec)
+new_rec = campaign.recommend(batch_quantity=5)
+```
+
+## Further specifictation of campaign
+
+Although only a search space and an objective are necessary to create a campaign,
+several other aspects can be changed by the user.
+1. **A strategy:**: By default, campaigns use the composite
```

Comment:
this can be reduced or turned into code. one design pattern i found quite useful its code like 

```python
bla = Campaign(
   required_param1='lorem',
   required_param2='ipsum',
   # now follow all optional parmameters with an explicit comment
    optional_param1 = 1337, # optional, 42 by default
   optional_param2 = strategy, # optional, for details see link_to_strategy
)
```

Reply:
this is an exact use case for pseudo code, eg I would not explicitly write down a `strategy` here in form of code

Reply:
Whether or not we can implement that when using a python environment depends on the decision that we take in the documentation testing PR, right?
For now, I have simply reduced the amount of text to the absolute minimum that I feel should be written here. Please resolve if happy.

Reply:
decision was "required changes should be made where its needed". But imo especially the `Campaign` user guide is probably THE page with sue case for pseudo code (because it is on highest level)